googletest/include/gtest/internal/gtest-port.h

```diff
@@ -2223,6 +2229,64 @@ const char* StringFromGTestEnv(const char* flag, const char* default_val);
 
 #endif  // !defined(GTEST_INTERNAL_DEPRECATED)
 
+#if GTEST_HAS_ABSL
+// Always use absl::any for UniversalPrinter<> specializations if googletest
+// is built with absl support.
+# define GTEST_INTERNAL_HAS_ANY 1
+#include "absl/types/any.h"
+namespace testing {
+namespace internal {
+using Any = ::absl::any;
+}  // namespace internal
+}  // namespace testing
+#else
+# ifdef __has_include
```

Comment:
I think this detection logic doesn't work for MSVC.
See https://github.com/abseil/abseil-cpp/blob/7853a7586c492ce8905c9e49f8679dada6354f2c/absl/base/config.h#L543
We can leave a TODO comment.

Reply:
I based that on the existing StringView detection as delivered in 6f5fd0d. If we wanna change this PR, then we should adjust the existing code.